breathecode/certificate/views.py

```diff
@@ -25,10 +26,38 @@
 
 @api_view(["GET"])
 @permission_classes([AllowAny])
-def get_specialties(request):
-    items = Specialty.objects.all()
-    serializer = SpecialtySerializer(items, many=True)
-    return Response(serializer.data, status=status.HTTP_200_OK)
+@capable_of("read_certificate")
+def get_academy_specialties(request, academy_id=None):
+    academy_id = request.headers.get("Academy") or request.headers.get("academy")
+    if not academy_id or not academy_id.isdigit():
+        return Response({"detail": "Valid academy ID is required"}, status=status.HTTP_400_BAD_REQUEST)
+    academy_id = int(academy_id)
```

Comment:
This is done by the decorator, you should get the academy_id from the parameters received in the function